### PR TITLE
ref(ts): Improve HoC typings

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Proptypes are now properly passed through all HoC components.

This fixed a few minor bugs that this exposed.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/sidebar/discover2Item.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ import withDiscoverSavedQueries from 'app/utils/withDiscoverSavedQueries';
 
 import SidebarItem from './sidebarItem';
 
-type Props = React.ComponentProps<SidebarItem> & {
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/static/sentry/app/components/sidebar/discover2Item.tsx]: I am working on this feature presently. I can take care of any errors if they come up.

**Comment by EvanPurkhiser** [comment]
Anything left on this?

File: src/sentry/static/sentry/app/components/acl/feature.tsx [hunk 2]
```diff
@@ -193,22 +190,4 @@ class Feature extends React.Component<FeatureProps> {
...
-    const features = this.props.config.features
-      ? Array.from(this.props.config.features)
-      : [];
-
-    return <Feature configFeatures={features} {...this.props} />;
-  }
-}
-
-export default withConfig(withOrganization(withProject(FeatureContainer)));
+export default withOrganization(withProject(withConfig(Feature)));
```
Comment (scefali) [inline src/sentry/static/sentry/app/components/acl/feature.tsx]: @EvanPurkhiser what's the reason for switching the order of the HoCs?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/acl/feature.tsx]: oops. This was me testing something actually, looks like I didn't put it back. Though in this case order isn't really important.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/acl/feature.tsx]: I can undo, but I don't really want to wait for tests again and it shouldn't affect anything

Comment (scefali) [reply src/sentry/static/sentry/app/components/acl/feature.tsx]: @EvanPurkhiser No worries, I was just curious

**Review by scefali — APPROVED** [review]

**Comment by scefali** [comment]
I may not be the best person to review this but seems good enough for me

**Review by markstory — APPROVED** [review]

**Pull request merged**